sound/soc/sof/sof-priv.h

```diff
@@ -54,10 +54,11 @@ extern int sof_core_debug;
 	(IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_ENABLE_DEBUGFS_CACHE) || \
 	 IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST))
 
-/* DSP D0ix sub-state */
-enum sof_d0_substate {
-	SOF_DSP_D0I0 = 0,	/* DSP default D0 substate */
-	SOF_DSP_D0I3,		/* DSP D0i3(low power) substate*/
+/* DSP power state */
+enum sof_dsp_power_state {
+	SOF_DSP_D0I0 = 0,
+	SOF_DSP_D0I3,	/* DSP low power D0 substate */
```

Comment:
if we want to make things generic then let's add D0i1, D0i1, D1 and D2!
There is no reason why we should only deal with those 3 values, the code should be extensible at will.

Reply:
@plbossart sorry but I dont understand the ask here. I cant tell if you are being serious or sarcastic. How can I make this generic? There are only 3 known states as of now and yes it could be extended in the future but thats unlikely to affect the current platforms. I could make this platform-specific if that sense.

As for your other suggestion about adding D0/D3 states and substates for each state, that would make sense if all states had substates but in our case its really only the D0 state that has them. 

So maybe I just completely missing your point here. Could you please spell it out for me?

Reply:
I was not being sarcastic.

D0, D1, D2, D3 are PCI/ACPI states. There is no reason to only list D0/D3. You can support only D0/D3 but there's no reason to assume they are the only states we will ever need.

The same is true for S0ix, in the past we supported S0i1 for audio, other platforms used S0i2.

Reply:
@plbossart are you suggesting we extend our device state definitions to match all those defined in the ACPI states and handle only the ones we care about?

And as far system states go, to use S0ix to cover all cases S0i1/2/3?

Reply:
yes, that was the suggestion. Would that be a problem?

Keep in mind that we *could* in theory use D0ix for low-power audio playback, aka 'deep-buffer', aka D0i3-streaming, where some of the memory windows remain powered to report timestamps. So it's in our best interest to allow for flexibility in the future with additional power states.

Reply:
Not a problem. I'm on it